src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -756,21 +756,14 @@ public unsafe void CopyTo(Array array, int index)
 
             if (array is int[] intArray)
             {
-                Div32Rem(m_length, out int extraBits);
+                int fullInts = Div32Rem(m_length, out int extraBits);
```

Comment:
There was missing a check here for destination array length. Should I add one?

Reply:
I would guess that bounds checking is delegated to `Array.Copy`. Is it causing undesirable behavior right now?

Reply:
Yeah, seems like it is throwing IndexOutOfRangeException (in 8.0.2) when doing:
```new bitArray(33).CopyTo(new int[1], 0)```
Documentation says it should throw `ArgumentException`

Reply:
Could you file a separate issue about it? Even though documentation says otherwise changing the exception type would be somewhat of a breaking change so it needs to be considered separately. It is likely that we need to investigate the `CopyTo` implementation in other collection types for similar discrepancies.